test/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepositoryTest.java

```diff
@@ -0,0 +1,313 @@
+        Assert.assertEquals(Period.seconds(TEST_PERIOD_IN_SECONDS), expected.get().getPeriod());
+        Assert.assertEquals(TEST_STATISTIC, expected.get().getStatistic());
+        Assert.assertEquals(TEST_QUANTITY, expected.get().getValue());
+        Assert.assertEquals(TEST_CONTEXT, expected.get().getContext());
+    }
+
+    @Test
+    public void addAlertWithNoExtension() {
+        final UUID uuid = UUID.randomUUID();
+        final Alert alert = alertBuilder
+                .setId(uuid)
+                .setExtensions(Collections.emptyMap())
+                .build();
+        alertRepo.addOrUpdateAlert(alert);
+        final Alert expectedAlert = alertRepo.get(uuid).get();
+        Assert.assertTrue(expectedAlert.getExtensions().isEmpty());
+    }
+
+    @Test(expected = PersistenceException.class)
+    public void testThrowsExceptionWhenQueryFails() {
+        final UUID uuid = UUID.randomUUID();
+        alertRepo.addOrUpdateAlert(alertBuilder
+                .setId(uuid)
+                .setCluster("new-cluster")
+                .build());
+        models.ebean.Alert ebeanAlert1 = Ebean.find(models.ebean.Alert.class)
+                .where()
+                .eq("uuid", uuid)
+                .findUnique();
+        models.ebean.Alert ebeanAlert2 = Ebean.find(models.ebean.Alert.class)
+                .where()
+                .eq("uuid", uuid)
+                .findUnique();
+        try (Transaction transaction = Ebean.beginTransaction()) {
+            ebeanAlert1.setCluster("new-cluster1");
+            ebeanAlert2.setCluster("new-cluster2");
+            ebeanAlert2.save();
+            ebeanAlert1.save();
+            transaction.commit();
+        } catch (final IOException e) {
+            //Do Nothing
+        }
+    }    private final DatabaseAlertRepository.AlertQueryGenerator queryGenerator = new DatabaseAlertRepository.GenericQueryGenerator();
+    private final DatabaseAlertRepository alertRepo = new DatabaseAlertRepository(queryGenerator);
+    private final DefaultAlert.Builder alertBuilder = new DefaultAlert.Builder()
+            .setId(UUID.randomUUID())
+            .setCluster(TEST_CLUSTER)
+            .setMetric(TEST_METRIC)
+            .setContext(TEST_CONTEXT)
+            .setService(TEST_SERVICE)
+            .setExtensions(TEST_EXTENSIONS)
+            .setName(TEST_NAME)
+            .setOperator(TEST_OPERATOR)
+            .setPeriod(Period.seconds(TEST_PERIOD_IN_SECONDS))
+            .setStatistic(TEST_STATISTIC)
+            .setValue(TEST_QUANTITY);
+
+    private static final String TEST_CLUSTER = "test-cluster";
+    private static final String TEST_METRIC = "test-metric";
+    private static final String TEST_SERVICE = "test-service";
+    private static final Context TEST_CONTEXT = Context.CLUSTER;
+    private static final Map<String, Object> TEST_EXTENSIONS = ImmutableMap.of(
+            "severity", "CRITICAL",
+            "notify", "abc@groupon.com",
```

Comment:
You shouldn't use a groupon address here. Please use example.com as the domain (for example).